src/Http.hs

```diff
@@ -17,13 +23,31 @@ import           Network.HTTP.Types
 import           Network.Wai.Handler.Warp
 import           Network.Socket
 
-start :: String -> IO (Bool, String) -> IO ()
-start socketName trigger = do
+socketName :: String
+socketName = ".autospec.sock"
+
+socketAddr :: SockAddr
+socketAddr = SockAddrUnix socketName
+
+newSocket :: IO Socket
+newSocket = socket AF_UNIX Stream 0
+
+withSocket :: (Socket -> IO a) -> IO a
+withSocket action = bracket newSocket sClose action
+
+withServer :: IO (Bool, String) -> IO a -> IO a
```

Comment:
I moved the thread spawning / killing into a separate function `withThread`.  Hope that helps.